subgraphs/inventory/app/src/main/resources/inventory.graphql

```diff
@@ -1,12 +1,14 @@
+extend schema @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key", "@external", "@requires", "@shareable"])
+
 type Product implements ProductItf @key(fields: "id") {
   id: ID!
   dimensions: ProductDimension @external
   delivery(zip: String): DeliveryEstimates @requires(fields: "dimensions { size weight }")
 }
 
 type ProductDimension {
-  size: String
-  weight: Float
+  size: String @external
```

Comment:
@martinbonnin @pcmanus is this `@external` needed when we have it on line 5 already?

Reply:
I would say so? These are different fields that could potentially be resolved differently? But don't take my word for it. I'll defer to @pcmanus on this one.

Reply:
Thanks for the in-depth explanation! 

FWIW, and as a Federation Newbie, I went with `@external` because I was under the impression that `@shareble` meant that the field is resolvable by the subgraph. Which is definitely not the case here.

So I understand the rationale behind the `@shareable` directive but could we be more explicit about the fact that the query planner will not try to resolve a `@shareable` field if it's only used through external ones? That might not be an easy concept to explain, I don't know. Just thinking out loud.

As far as this PR is concerned both work for me! I'm happy to switch to `@shareable` or leave as is.

Reply:
FWIW, @hwillson went for `@shareable` in the [federation-subgraph-compatibility](https://github.com/apollographql/apollo-federation-subgraph-compatibility/pull/82/files#diff-5b74cdeab4a2f4cf7b87c50f22bd1b1101d8869074d7dfea99cf138d4a2fcc40R18) repo so might as well just go with that for consistency.

Reply:
@martinbonnin  sounds good. with the inventory subgraph schema now using Fed 2 syntax, let's go with `@shareable` 👍

Reply:
Done. I also bump federation-jvm to alpha.4

Reply:
cc @StephenBarlow this is probably worth a mention in the Federation docs
> So I understand the rationale behind the @shareable directive but could we be more explicit about the fact that the query planner will not try to resolve a @shareable field if it's only used through external ones? That might not be an easy concept to explain, I don't know. Just thinking out loud.